src/components/modals/WcSmartContractModal.tsx

```diff
@@ -112,6 +112,7 @@ export const WcSmartContractModal = (props: Props) => {
           const spendInfo: EdgeSpendInfo = await wallet.otherMethods.txRpcParamsToSpendInfo(cleanPayload.params[0])
           const tx = await wallet.makeSpend(spendInfo)
           const signTx = await wallet.signTx(tx)
+          await wallet.saveTx(signTx)
```

Comment:
Can we put this below `walletConnect.approveRequest`? If WalletConnect fails, we don't want to call `saveTx` at all.

Reply:
good idea, done